perun-registrar-lib/src/test/java/cz/metacentrum/perun/registrar/RegistrarBaseIntegrationTest.java

```diff
@@ -794,6 +799,291 @@ private Application prepareApplicationToGroup(User user, Group group) {
+
+		assertThat(result.getData())
+			.hasSize(1);
+		assertThat(result.getData().get(0).getId())
+			.isEqualTo(voApplication1.getId());
+
+	}
+
+	@Test
+	public void getMembersPageOffsetWorks() throws Exception {
+		System.out.println("getMembersPageOffsetWorks");
+
+		// create groups
+		Group group1 = setUpGroup("Group1", "Cool folks");
+		// create users
+		User user1 = setUpUser("Joe", "Doe");
+		User user2 = setUpUser("Barney", "Stinson");
+
+		Application voApplication1 =  setUpApplicationGroup(user1, group1);
+		Application voApplication2 = setUpApplicationGroup(user2, group1);
+
+		ApplicationsPageQuery query = new ApplicationsPageQuery(1, 1, SortingOrder.ASCENDING, ApplicationsOrderColumn.ID, List.of("APPROVED"), true);
+
+
+		Paginated<Application> result = registrarManager.getApplicationsPage(session, vo, query);
+
+		assertThat(result.getData())
+			.hasSize(1);
+		assertThat(result.getData().get(0).getUser().getId())
+			.isEqualTo(voApplication2.getUser().getId());
+	}
+
+	@Test
+	public void getMembersPageIdSortWorks() throws Exception {
+		System.out.println("getMembersPageIdSortWorks");
+
+		// create groups
+		Group group1 = setUpGroup("Group1", "Cool folks");
+		// create users
+		User user1 = setUpUser("Joe", "Doe");
+		User user2 = setUpUser("Barney", "Stinson");
+
+		setUpApplicationGroup(user1, group1);
+		setUpApplicationGroup(user2, group1);
+
+		ApplicationsPageQuery query = new ApplicationsPageQuery(4, 0, SortingOrder.DESCENDING, ApplicationsOrderColumn.STATE, List.of("APPROVED", "VERIFIED"), true);
+
+
+		Paginated<Application> result = registrarManager.getApplicationsPage(session, vo, query);
+
+		List<Application.AppState> returnedAppStates = result.getData().stream()
+			.map(Application::getState).toList();
+
+		assertThat(returnedAppStates).containsExactly(Application.AppState.VERIFIED, Application.AppState.VERIFIED,
+			Application.AppState.APPROVED, Application.AppState.APPROVED);
+	}
+
+	@Test
+	public void getApplicationsPageBasedOnSearchString() throws Exception {
+		System.out.println("getApplicationsPageBasedOnSearchString");
+
+		// create groups
+		Group group1 = setUpGroup("Group1", "Cool folks");
+		// create users
+		User user1 = setUpUser("Joe", "Doe");
+		User user2 = setUpUser("Barney", "Stinson");
+
+		setUpApplicationGroup(user1, group1);
+		setUpApplicationGroup(user2, group1);
+
+		ApplicationsPageQuery query = new ApplicationsPageQuery(4, 0, SortingOrder.DESCENDING, ApplicationsOrderColumn.ID, "barn", List.of("APPROVED"), true);
+
+		Paginated<Application> result = registrarManager.getApplicationsPage(session, vo, query);
+
+		List<User> returnedUsers = result.getData().stream()
+			.map(Application::getUser).toList();
+
+		assertThat(returnedUsers).containsOnly(user2);
+	}
+
+	@Test
+	public void getApplicationsPageFindByUsername() throws Exception {
+		System.out.println("getApplicationsPageFindByGroupName");
```

Comment:
```suggestion
		System.out.println("getApplicationsPageFindByUsername");
```

Anyway, this test seems to do the same thing as the previous one, I'd say we can remove it.